### PR TITLE
Fix internal.d.ts

### PR DESCRIPTION
**Pull request by 38elements**

### CODE REVIEW

File: src/internal.d.ts [hunk 1]
```diff
@@ -10,12 +10,12 @@ export interface FunctionalComponent<P = {}> extends preact.FunctionalComponent<
...
 
 export interface PreactElement extends HTMLElement {
-	_prevVNode?: VNode<any>;
+	_prevVNode?: VNode<any> | null;
 	/** Event listeners to support event delegation */
 	_listeners: Record<string, (e: Event) => void>;
 
 	// Preact uses this attribute to detect SVG nodes
-	ownerSVGElement?: SVGElement;
+	ownerSVGElement?: SVGElement | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://developer.mozilla.org/en-US/docs/Web/API/SVGElement

File: src/internal.d.ts [hunk 2]
```diff
@@ -53,10 +53,10 @@ export interface Component<P = {}, S = {}> extends preact.Component<P, S> {
 	 * Pointer to the parent dom node. This is only needed for top-level Fragment
 	 * components or array returns.
 	 */
-	_parentDom?: PreactElement;
-	_prevVNode?: VNode;
+	_parentDom?: PreactElement | null;
+	_prevVNode?: VNode | null;
 	_ancestorComponent?: Component<any, any>;
-	_processingException?: Component<any, any>;
+	_processingException?: Component<any, any> | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://github.com/developit/preact/blob/40a696f4c565d5dbcfb4e1c90fffe91d1e125486/src/diff/index.js#L175

File: src/internal.d.ts [hunk 3]
```diff
@@ -53,10 +53,10 @@ export interface Component<P = {}, S = {}> extends preact.Component<P, S> {
 	 * Pointer to the parent dom node. This is only needed for top-level Fragment
 	 * components or array returns.
 	 */
-	_parentDom?: PreactElement;
-	_prevVNode?: VNode;
+	_parentDom?: PreactElement | null;
+	_prevVNode?: VNode | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://github.com/developit/preact/blob/40a696f4c565d5dbcfb4e1c90fffe91d1e125486/src/diff/index.js#L130

File: src/internal.d.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export interface VNode<P = {}> extends preact.VNode<P> {
 export interface Component<P = {}, S = {}> extends preact.Component<P, S> {
 	constructor: preact.ComponentFactory<P>;
 	state: S; // Override Component["state"] to not be readonly for internal use, specifically Hooks
-	base?: PreactElement;
+	base?: PreactElement | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://github.com/developit/preact/blob/40a696f4c565d5dbcfb4e1c90fffe91d1e125486/src/diff/index.js#L329

File: src/internal.d.ts [hunk 5]
```diff
@@ -53,10 +53,10 @@ export interface Component<P = {}, S = {}> extends preact.Component<P, S> {
 	 * Pointer to the parent dom node. This is only needed for top-level Fragment
 	 * components or array returns.
 	 */
-	_parentDom?: PreactElement;
-	_prevVNode?: VNode;
+	_parentDom?: PreactElement | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://github.com/developit/preact/blob/40a696f4c565d5dbcfb4e1c90fffe91d1e125486/src/diff/index.js#L329

File: src/internal.d.ts [hunk 6]
```diff
@@ -10,12 +10,12 @@ export interface FunctionalComponent<P = {}> extends preact.FunctionalComponent<
 export type ComponentFactory<P> = preact.ComponentConstructor<P> | FunctionalComponent<P>;
 
 export interface PreactElement extends HTMLElement {
-	_prevVNode?: VNode<any>;
+	_prevVNode?: VNode<any> | null;
```
Comment (38elements) [inline src/internal.d.ts]: https://github.com/developit/preact/blob/0c104d95e4265e7e1824e63c0bd6e0f391abd81a/src/render.js#L28

**Comment by andrewiggins** [comment]
@38elements Nice PR! Appreciate the dilengence to link to the relevant code ✔️

**Pull request merged**